### PR TITLE
Incorrect handling of UTF-8 characters in latex reference statement

### PR DESCRIPTION
**Pull request by albert-github**
In case in there is a literal utf-8 character  like the the Unicode En Dash (U+2013)  a pages is accessed with `\T1\endash` like:
```
https://en.wikipedia.org/wiki/Damerau\T1\textendash_Levenshtein_distance
```
and this page cannot be found, so the UTF-8 character has to be handled properly here.
This is found when solving issue #8241

Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/5680746/example.tar.gz)

### CODE REVIEW

File: src/util.cpp [hunk 1]
```diff
@@ -5309,7 +5309,63 @@ QCString latexFilterURL(const char *s)
...
+                unsigned char id = (unsigned char)ids[m];
+                t << "\\%" << map[id>>4] << map[id&0xF];
+              }
+              p += l - 1;
+            }
+          }
+          if (doEscape) // not a valid unicode char or escaping needed
+          {
+            static char map[] = "0123456789ABCDEF";
+            unsigned char id = (unsigned char)c;
```
Comment (doxygen) [inline src/util.cpp:5361]: `map` and `id` are not used in this case and nothing is escaped. Looks like something missing for this case...

Comment (albert-github) [reply src/util.cpp:5361]: Copy paste error, we don't have an idea how to escape these characters here so so write the as is, the lines:
```
     static char map[] = "0123456789ABCDEF";
     unsigned char id = (unsigned char)c;
```
should not be present.

Comment (doxygen) [reply src/util.cpp:5361]: ok, I think also the name of the `doEscape` variable is confusing. If it is `false' % escaping is done, so exactly the opposite of what the name suggests. Should this be corrected as well or am I missing something?

Comment (albert-github) [reply src/util.cpp:5361]: Most likely yes, I copied the code from another part and (unfortunately) didn't pay attention to the names of all variables.

**Pull request merged**